src/shogun/machine/LinearMulticlassMachine.h

```diff
@@ -64,11 +64,6 @@ class LinearMulticlassMachine : public MulticlassMachine
 		void set_features(std::shared_ptr<DotFeatures> f)
 		{
 			m_features = f;
-			for (auto m: m_machines)
-			{
-				auto machine = m->as<LinearMachine>();
-				machine->set_features(f);
-			}
```

Comment:
how are the features now passed to `LinearMachine`? Is this done in `LinearMulticlassMachine`'s train routine?

Reply:
yes, it should be done in `LinearMulticlassMachine`'s `train`, but the `train` is defined in `MulticlassMachine` if I change the definition of `train`, there will be a lot of class to be changed, so I plan to do this when I refactor `MulticlassMachine`.

Reply:
I see, but don't the tests break?